### PR TITLE
Day night masking sphinx documentation

### PR DESCRIPTION
**Pull request by kperrynrel**
## Description

This PR adds data sets for NREL's SERF east installation, and runs day-night masking on 1-minute sampled and 15-minute sampled data. This example is for the Sphinx gallery. Related to https://github.com/pvlib/pvanalytics/issues/133.

## Checklist

The following items must be addressed before the code can be merged. 
Please don't hesitate to ask for help if you are unsure of how to accomplish any of the items. 
*You are free to remove any checklist items that do not apply or add additional items that are 
not on this list*

- [X] Clearly documented all new API functions with [PEP257](https://www.python.org/dev/peps/pep-0257/) and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings
- [x] Adds description and name entries in the appropriate "what's new" file 
      in [`docs/whatsnew`](https://github.com/pvlib/pvanalytics/tree/master/docs/whatsnew) 
      for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` 
      or this Pull Request with `` :pull:`num` ``. Includes contributor name 
      and/or GitHub username (link with `` :ghuser:`user` ``).
- [X] Non-API functions clearly documented with docstrings or comments as necessary
- [x] Pull request is nearly complete and ready for detailed review

### CODE REVIEW

File: docs/examples/day-night-masking.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+plt.ylabel("AC Power (kW)")
+plt.tight_layout()
+plt.show()
+
+# %%
+# The pvlib.solarposition.sun_rise_set_transit_spa function is
+# used to get sunrise and sunset times for each day at the site location, and
+# the daytime mask is calculated based on these times. Data associated with
+# day time periods is labeled as True, and data associated with nighttime
+# periods is labeled as False.
```
Comment (kanderso-nrel) [inline docs/examples/day-night-masking.py:51]: Probably worth pointing out that this is only done for comparison; "true" sunrise and sunset times are not needed to run `pvanalytics.features.daytime.power_or_irradiance`.  Could even move the SPA calculation and put it after the plot.

File: docs/examples/day-night-masking.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+# Plot the time series.
+data['ac_power__752'].plot()
+plt.xticks(rotation=20)
+plt.xlabel("Date")
+plt.ylabel("AC Power (kW)")
+plt.tight_layout()
+plt.show()
+
+# %%
+# The pvlib.solarposition.sun_rise_set_transit_spa function is
```
Comment (kanderso-nrel) [inline docs/examples/day-night-masking.py:47]: Could use `:py:func:` here to link out to the pvlib docs

File: docs/examples/day-night-masking.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+data['sunrise_time'] = sunrise_sunset_df['sunrise']
+data['sunset_time'] = sunrise_sunset_df['sunset']
+
+data['daytime_mask'] = True
+data.loc[(data.index < data.sunrise_time) |
+         (data.index > data.sunset_time), "daytime_mask"] = False
+
+# %%
+# Set all negative values in the AC power time series to 0.
+data.loc[data['ac_power__752'] < 0, 'ac_power__752'] = 0
```
Comment (kanderso-nrel) [inline docs/examples/day-night-masking.py:66]: I think this comment should mention the purpose -- is this necessary for reliable classifications?

File: docs/examples/day-night-masking.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+data.loc[predicted_day_night_mask, 'ac_power__752'].plot(ls='', marker='o')
+data.loc[~predicted_day_night_mask, 'ac_power__752'].plot(ls='', marker='o')
+sunrise_sunset_times = sunrise_sunset_df[['sunrise',
+                                          'sunset']].drop_duplicates()
+for sunrise, sunset in sunrise_sunset_times.itertuples(index=False):
+    plt.axvline(x=sunrise, c="blue")
+    plt.axvline(x=sunset, c="red")
+plt.legend(labels=["AC Power", "Daytime", "Nighttime",
+                   "SPA Sunrise", "SPA Sunset"])
+plt.xticks(rotation=20)
```
Comment (kanderso-nrel) [inline docs/examples/day-night-masking.py:84]: I feel silly for complaining about this, but when I saw the tilted labels in the plot PNGs my first thought was that something somehow went wrong in the sphinx build and it was a rendering error.  Any good reason to apply a rotation here?

**Review by kanderso-nrel — COMMENTED** [review]
I'm not sure I see much value in running the same code again on 15 minute data.  I could support it if it was tweaking the optional parameters to get better results or something, but if it's essentially the same thing just with a different accuracy score, I'd lean towards removing it to keep the example more digestible.

File: docs/examples/day-night-masking.py (hunk 4 above)
Comment (kperrynrel) [reply docs/examples/day-night-masking.py:84]: image
Removed the tilt on the axes and things look good

File: docs/examples/day-night-masking.py (hunk 2 above)
Comment (kperrynrel) [reply docs/examples/day-night-masking.py]: Updated in commit https://github.com/pvlib/pvanalytics/pull/139/commits/8fca43cd2428c7d279072138e79cdd7043e47b62

File: docs/examples/day-night-masking.py (hunk 1 above)
Comment (kperrynrel) [reply docs/examples/day-night-masking.py]: @kanderso-nrel I added some additional comments on this being for comparison. I did move the script around a little, but this is still before the plot as we use SPA sunrise and sunset in the plot (they are represented by the vertical lines)

File: docs/examples/day-night-masking.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+# with SPA nighttime periods is labeled as False.
+# SPA sunrise and sunset times are used here as a point of comparison to the
+# sensor-based AC power data; SPA-based sunrise and sunset values are not
+# needed to run :py:func:`pvanalytics.features.daytime.power_or_irradiance`.
+
+sunrise_sunset_df = pvlib.solarposition.sun_rise_set_transit_spa(data.index,
+                                                                 latitude,
+                                                                 longitude,
+                                                                 how='numpy',
+                                                                 delta_t=67.0)
```
Comment (cwhanse) [inline docs/examples/day-night-masking.py:75]: I'd delete the optional kwargs, no reason to suggest to a user of this example that values other than the default are needed.

**Review by cwhanse — APPROVED** [review]
Thanks @kperrynrel

**Pull request merged**